v0/destinations/hs/transform.js

```diff
@@ -91,30 +91,48 @@ const processRouterDest = async inputs => {
     const respEvents = getErrorRespEvents(null, 400, "Invalid event array");
```

Comment:
@sanpj2292 this might be OBE and good to be refactored after we close [the other refactoring PR](https://github.com/rudderlabs/rudder-transformer/pull/1338/files#diff-6dc65abd2e86ec4df04756af6dd5d01dabf4a3fcf1d8fd8c880323eb513770cc).

@utsabc @akashrpo do we have any pressing need for these changes?

Reply:
Yes we need this for any incorrect creds this will result in default 500 as we are not explicitly throwing error and causing rt pile up

Reply:
I see the original PR is merged now. @akashrpo refactor these changes accordingly.

Reply:
Done

Reply:
@saikumarrs can you please approve the PR?